File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        vector<int> res(nums.size(), 0);
        int l = 0, r = nums.size() - 1, x = r;
        while (l <= r) {
            if (pow(nums[l], 2) > pow(nums[r], 2)) {
                res[x] = pow(nums[l], 2);
                ++l;
            }
            else {
                res[x] = pow(nums[r], 2);
                --r;
            }
            --x;
        };
        return res;
    };
};

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        negSq = [i**2 for i in nums if i < 0]
        nums = nums[len(negSq):]

        if len(nums) > 0:
            i = 0
            nums[i] **= 2
            
            while len(negSq) > 0 and nums[i] > negSq[-1]:
                nums = nums[:i] + [negSq.pop()] + nums[i:]
                i += 1
                
            while i < len(nums) - 1:
                j = i + 1
                nums[j] **= 2

                while len(negSq) > 0 and negSq[-1] <= nums[j]:
                    nums = nums[:j] + [negSq.pop()] + nums[j:]
                    i += 1
                    j += 1
                i += 1

            if len(negSq) > 0:
                negSq.reverse()
                nums = nums + negSq

        else:
            negSq.reverse()
            return negSq
        
        return nums